apps/onboarding/src/wallet/wallet.service.ts

```diff
@@ -42,17 +41,65 @@ export class WalletService {
     private readonly cfg: AppConfig
   ) {}
 
+  async isAllowedMetaTransaction(
+    transaction: RawTransaction,
+    allowedTransactions: TxFilter[]
+  ): Promise<Result<true, MetaTxValidationError>> {
+    const metaTxDecode = this.decodeMetaTransaction(transaction)
+    if (metaTxDecode.ok === false) {
+      return metaTxDecode
+    }
+
+    const metaTx = metaTxDecode.result
+    const normDestination = normalizeAddress(metaTx.destination)
+
+    const txWithMatchingDestination = allowedTransactions.filter(
+      (tx) =>
+        normalizeAddress(tx.destination) === normDestination
+    )
+
+    if (txWithMatchingDestination.length === 0) {
+      return Err(new InvalidDestination(metaTx.destination))
+    }
+
+    const metaTxMethodId = trimLeading0x(metaTx.data).slice(0, 8)
+    const matchingTx = txWithMatchingDestination.find(
+      (tx) => {
+        const methodId = trimLeading0x(tx.methodId)
+        return metaTxMethodId === methodId
```

Comment:
Any normalization required before comparison? (ex. toLowerCase)

Reply:
Currently no, but I guess it's worth adding just to be sure that nothing changes somewhere down the line at any point and this breaks.